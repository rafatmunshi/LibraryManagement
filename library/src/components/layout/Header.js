import React from 'react';

const Header = () => {
    return ( <header style = { headerStyle } >
        <h1> Library Management System </h1>
        View Books in the Library </header>
    )
};

const headerStyle = {
    background: "#333",
    color: '#fff',
    textAlign: 'center',
    padding: '10px'
};

export default Header;