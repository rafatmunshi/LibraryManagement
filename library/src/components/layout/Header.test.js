import toJson from 'enzyme-to-json';
import React from "react";
import Header from './Header';
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("<Header />", () => {
  let wrapperForShallowCopy;
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, "useState");
  useStateSpy.mockImplementation((init) => [init, setState]);

  beforeEach(() => {
    wrapperForShallowCopy = Enzyme.shallow(<Header />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

it('should render correctly as we know it', () => {
  const wrapperForShallowCopy = shallow(<Header />)
  expect(toJson(wrapperForShallowCopy)).toMatchSnapshot();
});

});
