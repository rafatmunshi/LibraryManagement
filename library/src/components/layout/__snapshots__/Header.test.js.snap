// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly enzyme 1`] = `
<header
  style={
    Object {
      "background": "#333",
      "color": "#fff",
      "padding": "10px",
      "textAlign": "center",
    }
  }
>
  <h1>
     Library Management System 
  </h1>
  View Books in the Library 
</header>
`;

exports[`<Header /> should render correctly as we know it 1`] = `
<header
  style={
    Object {
      "background": "#333",
      "color": "#fff",
      "padding": "10px",
      "textAlign": "center",
    }
  }
>
  <h1>
     Library Management System 
  </h1>
  View Books in the Library 
</header>
`;
